quinn-proto/src/connection/mod.rs

```diff
@@ -2268,7 +2345,13 @@ impl Connection {
                             packet.header.is_1rtt(),
                         );
                     }
-                    self.process_decrypted_packet(now, remote, number, packet)
+
+                    let result = self.process_decrypted_packet(now, remote, number, packet);
+
+                    #[cfg(feature = "qlog")]
+                    self.emit_qlog_recovery_metrics(now);
```

Comment:
If the intent is to call this only after a datagram is fully processed, we should probably lift it into `handle_event`, after the call to `handle_coalesced`, just like how the other callsite added in this PR follows the `handle_coalesced` call in `handle_first_packet`. As is, this might get called multiple times for one datagram, although that probably isn't harmful.